test/e2e/framework/service_util.go

```diff
@@ -1450,3 +1461,50 @@ func GetServiceLoadBalancerCreationTimeout(cs clientset.Interface) time.Duration
 	}
 	return LoadBalancerCreateTimeoutDefault
 }
+
+func TestAffinity(jig *ServiceTestJig, execPod *v1.Pod, targetIp string, targetPort int, shouldHold bool) bool {
+	var hostname, response string
+	var err error
```

Comment:
remove this and try to have all err vars scoped as close as possible to their uses
better is to use the if scope
```
if err := foo(); err != nil {
...
}
```

Reply:
Fixed.